src/Conveyor/curved_belt_conveyor.gd

```diff
@@ -166,6 +164,577 @@ func _ready() -> void:
+			idx, idx + 2, idx + 3)
+		add_double_sided_triangle.call(surfaces.sides.vertices, surfaces.sides.indices, 
+			idx, idx + 3, idx + 1)
+
+	# 7. Add outer bottom transition from diagonal to metal lip
+	var outer_bottom_flat_to_metal_start_idx = surfaces.sides.vertices.size()
+	for data in outer_bottom_flat_to_metal_vertices:
+		surfaces.sides.vertices.append_array([data[0], data[1]])
+		surfaces.sides.normals.append_array([data[2], data[2]])
+		surfaces.sides.uvs.append_array([data[3], data[4]])
+
+	for i in range(segments):
+		var idx = outer_bottom_flat_to_metal_start_idx + (i * 2)
+		add_double_sided_triangle.call(surfaces.sides.vertices, surfaces.sides.indices, 
+			idx, idx + 2, idx + 3)
+		add_double_sided_triangle.call(surfaces.sides.vertices, surfaces.sides.indices, 
+			idx, idx + 3, idx + 1)
+
+	surfaces.sides.vertex_count = surfaces.sides.vertices.size()
+
+	for edges in [belt_edges.top_outer, belt_edges.bottom_outer]:
+		for i in range(segments):
+			var idx = i * 2
+			add_double_sided_triangle.call(edges.vertices, edges.indices, idx, idx + 1, idx + 3)
+			add_double_sided_triangle.call(edges.vertices, edges.indices, idx, idx + 3, idx + 2)
+
+	# Add all belt edges to the appropriate surfaces
+	for edge_name in belt_edges.keys():
+		var edge = belt_edges[edge_name]
+		var start_idx = surfaces.top.vertices.size() if edge_name.begins_with("top") else surfaces.bottom.vertices.size()
+		var target_surface = surfaces.top if edge_name.begins_with("top") else surfaces.bottom
+		
+		target_surface.vertices.append_array(edge.vertices)
+		target_surface.normals.append_array(edge.normals)
+		target_surface.uvs.append_array(edge.uvs)
+		
+		for i in range(edge.indices.size()):
+			target_surface.indices.append(start_idx + edge.indices[i])
+
+	surfaces.sides.vertex_count = surfaces.sides.vertices.size()
+	
+	# Add surfaces to mesh
+	for pair in [["top", 0], ["bottom", 1], ["sides", 2]]:
+		var surface_name = pair[0]
+		var surface_index = pair[1]
+		var surface = surfaces[surface_name]
+
+		# Skip empty surfaces
+		if surface.vertices.size() == 0:
+			continue
+
+		var arrays = []
+		arrays.resize(Mesh.ARRAY_MAX)
+		arrays[Mesh.ARRAY_VERTEX] = surface.vertices
+		arrays[Mesh.ARRAY_NORMAL] = surface.normals
+		arrays[Mesh.ARRAY_TEX_UV] = surface.uvs
+		arrays[Mesh.ARRAY_INDEX] = surface.indices
+		mesh.add_surface_from_arrays(Mesh.PRIMITIVE_TRIANGLES, arrays)
+
+		# Apply appropriate material based on surface
+		var material_to_use = belt_material
+		if surface_name == "sides":
+			material_to_use = metal_material
+		mesh.surface_set_material(surface_index, material_to_use)
+		
+	# Assign mesh to MeshInstance3D
+	curved_mesh.mesh = mesh
+
+	# Create collision shape
+	var collision_shape = $StaticBody3D.get_node_or_null("CollisionShape3D") as CollisionShape3D
+	if not collision_shape:
+		collision_shape = CollisionShape3D.new()
+		collision_shape.name = "CollisionShape3D"
+		$StaticBody3D.add_child(collision_shape)
+		collision_shape.owner = self
```

Comment:
This collision shape node should already exist too. No need to create a new one.